scenarios/test_codegen/test/lib_tests/EntityHistory_test.res

```diff
@@ -0,0 +1,222 @@
+open RescriptMocha
+
+type testEntity = {
+  id: string,
+  fieldA: int,
+  fieldB: option<string>,
+}
+
+let testEntitySchema: S.t<testEntity> = S.schema(s => {
+  id: s.matches(S.string),
+  fieldA: s.matches(S.int),
+  fieldB: s.matches(S.option(S.string)),
+})
+
+let testEntityRowsSchema = S.array(testEntitySchema)
+
+type testEntityHistory = Entities.EntityHistory.historyRow<testEntity>
+let testEntityHistorySchema = Entities.EntityHistory.makeHistoryRowSchema(testEntitySchema)
+
+let mockEntityTable = Table.mkTable(
+  "TestEntity",
+  ~fields=[
+    Table.mkField("id", Text, ~isPrimaryKey=true),
+    Table.mkField("fieldA", Integer),
+    Table.mkField("fieldB", Text, ~isNullable=true),
+  ],
+)
+let mockEntityHistory = Entities.EntityHistory.fromTable(mockEntityTable)
+
+let batchSetMockEntity = Table.PostgresInterop.makeBatchSetFn(
+  ~table=mockEntityTable,
+  ~rowsSchema=testEntityRowsSchema,
+)
+
+let getAllMockEntity = sql =>
+  sql
+  ->Postgres.unsafe(`SELECT * FROM "public"."${mockEntityTable.tableName}"`)
+  ->Promise.thenResolve(json => json->S.parseOrRaiseWith(testEntityRowsSchema))
+
+let getAllMockEntityHistory = sql =>
+  sql->Postgres.unsafe(`SELECT * FROM "public"."${mockEntityHistory.table.tableName}"`)
+
+describe("Entity history serde", () => {
+  it("serializes and deserializes correctly", () => {
```

Comment:
This test explains the desired effect/approach of using the schema for an entity